File: portfolio/src/util/Project.ts
interface ProjectInterface {
    projectName: string;
    description: string;
    tech: string[];
    git: string | null;
    live: string | null;
    image: any;
}
class Project implements ProjectInterface {
    projectName: string;
    description: string;
    tech: string[];
    git: string | null;
    live: string | null;
    image: any;
    constructor(projectName: string, description: string, tech: string[], git: string | null, live: string | null, image: any) {
        this.projectName = projectName;
        this.description = description;
        this.tech = tech;
        this.git = git;
        this.live = live;
        this.image = image;
    }
}

File: server/models/Project.py


class Project:
    def __init__(self, project_name, description, tech, git, live, image):
        self.project_name = project_name
        self.description = description
        self.tech = tech
        self.git = git
        self.live = live
        self.image = image
        return

    def getData(self):
        return {
            "projectName": self.project_name,
            "description": self.description,
            "tech": self.tech,
            "git": self.git,
            "live": self.live,
            "image": self.image,
        }
